src/test/java/core/basesyntax/service/FruitTransactionParserTest.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax.service;
+
+import static org.hamcrest.CoreMatchers.is;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.FruitTransactionParserImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionParserTest {
+    private static final String INFO_LINE = "type,fruit,quantity";
+    private static final String FIRST_TRANSACTION_LINE = "b,apple,50";
+    private static final String SECOND_TRANSACTION_LINE = "p,apple,40";
+    private static final String FIRST_INCORRECT_TRANSACTION_LINE = "p.orange;100";
+    private static final String SECOND_INCORRECT_TRANSACTION_LINE = "";
+    private static final String THIRD_INCORRECT_TRANSACTION_LINE = null;
+    private static List<String> firstIncorrectData;
+    private static List<String> secondIncorrectData;
+    private static List<String> thirdIncorrectData;
+    private static FruitTransaction firstTransaction;
+    private static FruitTransaction secondTransaction;
+    private static FruitTransactionParser fruitTransactionParser;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitTransactionParser = new FruitTransactionParserImpl();
+        firstTransaction = new FruitTransaction();
+        secondTransaction = new FruitTransaction();
+        firstIncorrectData = new ArrayList<>();
+        firstIncorrectData.add(INFO_LINE);
+        firstIncorrectData.add(FIRST_INCORRECT_TRANSACTION_LINE);
+        secondIncorrectData = new ArrayList<>();
+        secondIncorrectData.add(INFO_LINE);
+        secondIncorrectData.add(SECOND_INCORRECT_TRANSACTION_LINE);
+        thirdIncorrectData = new ArrayList<>();
+        thirdIncorrectData.add(INFO_LINE);
+        thirdIncorrectData.add(THIRD_INCORRECT_TRANSACTION_LINE);
+        firstTransaction.setFruit(new Fruit("apple"));
+        firstTransaction.setAmount(50);
+        firstTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        secondTransaction.setFruit(new Fruit("apple"));
+        secondTransaction.setAmount(40);
+        secondTransaction.setOperation(FruitTransaction.Operation.PURCHASE);
+    }
+
+    @Test
+    public void parse_inputEmptyList_notOk() {
+        try {
```

Comment:
No need to check every single letter in all your exceptions messages. It would be enough to check that exception was thrown and type of it. You can use this annotation:

@ Test(expected = RuntimeException.class)
public void parse_inputEmptyList_notOk() {